src/WebJobs.Script.WebHost/Security/KeyManagement/DefaultSecretManagerProvider.cs

```
@@ -93,17 +93,17 @@ internal ISecretsRepository CreateSecretsRepository()
                                                                  siteSlotName,
                                                                  _loggerFactory.CreateLogger<BlobStorageSasSecretsRepository>(),
                                                                  _environment,
-                                                                 _azureStorageProvider);
+                                                                 _azureBlobStorageProvider);
             }
-            else if (_azureStorageProvider.ConnectionExists(ConnectionStringNames.Storage))
+            else if (StorageConnectionExists())
```

Instead of doing a naïve check for the storage connection, should this call `_azureBlobStorageProvider.TryCreateHostingBlobContainerClient(out _)` to see if the storage connection can be used correctly? This is the approach in HostIdValidator (changed in this PR).

---

Sure I guess - things will fail pretty much right away anyways since the repository will attempt to create a client

---

Right, things will fail right away when we do use the SecretManager, but the problem is that we decide on the provider in this class here. If it's a broken storage connection, we may want to use throw the InvalidOperationException immediately rather than creating a SecretManager and then always fail to use it. I will change this as proposed.